src/coffee/sync/utils/product.coffee

```diff
@@ -92,6 +92,12 @@ class ProductUtils extends BaseUtils
     actions = []
     _.each actionsBaseList(), (item) =>
       action = @_buildBaseAttributesAction(item, diff, old_obj)
+      if action && action.action == "setCategoryOrderHint"
+        if _.isUndefined action.categoryOrderHints
+          return
+        action.categoryId = Object.keys(action.categoryOrderHints)[0]
```

Comment:
As long as the `_buildBaseAttributesAction` method works there will be a first element. Otherwise there would not be a diff in the `categoryOrderHints`, in which case we would not get here anyways.
@hajoeichler Should I add some safety checks anyways?